drizzle-orm/src/mysql-core/functions/aggregate.ts

```diff
@@ -0,0 +1,149 @@
+import { is, entityKind } from '~/entity.ts';
+import { MySqlColumn } from '../columns/index.ts';
+import { sql, type SQLWrapper, isSQLWrapper, type SQLChunk } from '~/sql/index.ts';
+import { MySqlBuiltInFunction } from './common.ts';
+import { type MaybeDistinct, getValueWithDistinct } from '~/distinct.ts';
+
+export class MySqlAggregateFunction<T = unknown> extends MySqlBuiltInFunction<T> {
+	static readonly [entityKind]: string = 'MySqlAggregateFunction';
+}
+
+/**
+ * Returns the number of values in `expression`.
+ *
+ * ## Examples
+ *
+ * ```ts
+ * // Number employees with null values
+ * db.select({ value: count() }).from(employees)
+ * // Number of employees where `name` is not null
+ * db.select({ value: count(employees.name) }).from(employees)
+ * // Number of employees where `name` is distinct (no duplicates)
+ * db.select({ value: count(distinct(employees.name)) }).from(employees)
+ * ```
+ */
+export function count<T extends 'number' | 'bigint' | undefined = undefined>(expression?: MaybeDistinct<SQLWrapper> | '*', config?: {
```

Comment:
I would think that if this function would just return for example:
```ts
sql`count(${expression})`.mapWith(config?.mode === 'number' ? Number : BigInt)
```
The beautiful part is that when the dialect is constructing the query in calls `mergeQueries` recursively. So you could even do:
```ts
count(distinct(users.name))
```
And I would still construct the query 
```sql
count(distinct(`users`.`name`))`
```
in MySql. Assuming you construct `distinct` the same way.

Reply:
`count(distinct(expression))` syntax has been removed in favor of `countDistinct(expression)`.